refl1d/probe.py

```diff
@@ -180,7 +239,9 @@ def __init__(self, T=None, dT=0, L=None, dL=0, data=None,
                  intensity=1, background=0, back_absorption=1, theta_offset=0,
                  sample_broadening=0,
                  back_reflectivity=False, name=None, filename=None,
-                 dQ=None, resolution='normal'):
+                 dQ=None, resolution='normal',
+                 oversampling: Optional[Union[float, Oversampling]] = None,
+                 critical_edge_oversampling: Optional[CriticalEdgeOversampling] = None):
```

Comment:
Need to add oversampling/critical_edge_oversampling to the doc strings.

I'm not too keen on the interface for oversampling:
```
Probe(..., oversampling={'n': 5, 'seed': 12}, ...)
```

Shall we drop the sampling and just use a simple `n=...` selected linearly between the neighbouring points? Or maybe set some maximum δQ out and make sure there is coverage out to ±3ΔQ? Instead of δQ maybe make it a boolean and set δQ = 2π/5w where w is the total thickness of the sample? Instead of specifying oversampling, look at the theory function and apply it if the sample is thick and the resolution is bigger than δQ?

Similarly for critical edge oversampling `substrate` and `surface` can come from the model, `n` can come from the model thickness and `delta` can come from the resolution. To get the info we need from the model change Q_calc to a function which takes these details as parameters.

Probably more work than you have time for right now.

In lieu of a rewrite, it might be nice to allow the user to hand over the substrate/surface materials rather than making these fixed constants.

Reply:
I think going the other direction might work even better... Just have the user specify the Qc, which they can read right off a plot?